### PR TITLE
Modified the configuration file handling system

### PR DESCRIPTION
**Pull request by kshitij10496**
- [x] Assigned the entire specified configuration to `config` rather than a few key-value pairs.
- [x] Personalised all the messages.
- [x] Remove base `except` clause.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -103,23 +103,22 @@ def main():
...
-                                "message"][act][pos].replace("{name}", author)
-                            config["message"][act][pos] = new_config[
-                                "message"][act][pos].replace("{name}", author)
-                            config["message"][act][pos] = new_config[
-                                "message"][act][pos].replace("{name}", author)
-                    config["scanner"]["diff_only"] = new_config["scanner"]["diff_only"]
-            except:  # Bad yml file
-                pass
+                    # overriding the default configuration with the one specified
+                    config = new_config
```
Comment (OrkoHunter) [inline app.py]: Overriding the `config` dictionary might cause problems in [this](https://github.com/kshitij10496/pep8speaks/blob/94b43fcfe551a7a9321c906b13b3edd40bf098bb/app.py#L179-L188) block. Suppose the new config only has the `ignore` key-value pairs.

File: app.py [hunk 2]
```diff
@@ -103,23 +103,22 @@ def main():
...
+                except yaml.YAMLError as e:  # Bad YAML file
+                    pass
+
+            # personalising the messages
+            if "message" in config:
+                for act in config["message"].keys():
+                    # can be either "opened" or "updated"
+                    for pos in act.keys():
+                        # can be either "header" or "footer"
+                        config["message"][act][pos].replace("{name}", author)
```
Comment (OrkoHunter) [inline app.py]: I can't remember what the hell was I doing repeating the same line four times. This is nice!

File: app.py (hunk 1 above)
Comment (kshitij10496) [reply app.py]: I agree.

File: app.py (hunk 2 above)
Comment (kshitij10496) [reply app.py]: I guess you were trying to perform this operation by brute-force on 4 times.

File: app.py [hunk 3]
```diff
@@ -103,23 +103,23 @@ def main():
...
+                        config[key] = new_config[key]
+
+                except yaml.YAMLError as e:  # Bad YAML file
+                    pass
+
+            # personalising the messages
+            if "message" in config:
+                for act in config["message"]:
+                    # can be either "opened" or "updated"
+                    for pos in act:
```
Comment (OrkoHunter) [inline app.py]: This should be
```
for pos in config["message"][act]:
```

Comment (kshitij10496) [reply app.py]: You are right about this. I must have overlooked it somehow.

File: app.py [hunk 4]
```diff
@@ -103,23 +103,23 @@ def main():
...
+                except yaml.YAMLError as e:  # Bad YAML file
+                    pass
+
+            # personalising the messages
+            if "message" in config:
+                for act in config["message"]:
+                    # can be either "opened" or "updated"
+                    for pos in config["message"][act]:
+                        # can be either "header" or "footer"
+                        config["message"][act][pos].replace("{name}", author)
```
Comment (OrkoHunter) [inline app.py]: `"".replace()` returns a new string but does not change the original string.
So, we need to update the dictionary with this new value.

Comment (kshitij10496) [reply app.py]: Done

Comment (OrkoHunter) [reply app.py]: It's not updated here. :confused:

**Comment by kshitij10496** [comment]
Woah ! There seems to be some bug.

**Comment by OrkoHunter** [comment]
I'll delete @pep8speaks's last two comments and re-open this PR after closing.

**Comment by OrkoHunter** [comment]
Woah, it just stopped working for the config file.

**Comment by kshitij10496** [comment]
Yes and that too all of a sudden.
With no changes to the `master` !!!

**Comment by OrkoHunter** [comment]
@kshitij10496 Working now. Didn't do a single damn thing. Just restarted the app. Maybe the `.pep8speaks.yml` file got stuck there.

**Comment by OrkoHunter** [comment]
This looks good. Merging now.

**Comment by kshitij10496** [comment]
That's great.
But I still don't understand what caused this behaviour.
"Reveries" ? ;)

File: app.py [hunk 5]
```diff
@@ -103,23 +103,25 @@ def main():
...
+            if "message" in config:
+                for act in config["message"]:
+                    # can be either "opened" or "updated"
+                    for pos in config["message"][act]:
+                        # can be either "header" or "footer"
+                        msg = config["message"][act][pos]
+                        new_msg = msg.replace("{name}", author)
+                        config["message"][act][pos] = new_msg
 
             if PEP8SPEAK_YML_FOUND:
```
Comment (debugger22) [inline app.py:126]: Better use `PEP8SPEAKS_YML_FOUND`

Comment (OrkoHunter) [reply app.py:126]: Yeah noticed it now. Updated.

**Pull request merged**